lib/gdpr/gateway/user_details.rb

```diff
@@ -2,37 +2,37 @@
 
 class Gdpr::Gateway::Userdetails
   SESSION_BATCH_SIZE = 500
+
+  def initialize
+    @logger = Logger.new($stdout)
+  end
+
   def delete_users
-    logger = Logger.new($stdout)
+    @logger.info("Finding users that have been inactive for 12 months")
+    inactive_users = find_inactive_users(12)
 
-    logger.info("Finding users that have been inactive for 12 months")
-    inactive_users = WifiUser::User.where(
-      Sequel.lit("
-      (last_login < DATE_SUB(NOW(), INTERVAL 12 MONTH)
-      OR (last_login IS NULL AND created_at < DATE_SUB(NOW(), INTERVAL 12 MONTH)))
-      AND username != 'HEALTH'
-      "),
-    ).all
+    @logger.info("Found #{inactive_users.size} users that have been inactive for 12 months")
 
-    logger.info("Found #{inactive_users.size} users that have been inactive for 12 months")
+    @logger.info("Notifying users they have been removed from GovWifi")
 
-    logger.info("Notifying user they have been removed from GovWifi")
     inactive_users.each do |user|
       username = user[:username]
       contact = user[:contact]
 
       if user.mobile?
         WifiUser::SMSSender.notify_user(username, contact)
-        logger.info("SMS sent to #{username}: #{contact}")
       elsif user.valid_email?(contact)
-        WifiUser::EmailSender.notify_user(username, contact)
-        logger.info("Email sent to #{username}: #{contact}")
+        begin
+          WifiUser::EmailSender.notify_user(username, contact)
+        rescue Notifications::Client::BadRequestError => e
+          handle_email_error(e, username, contact)
```

Comment:
we may want to rescue StandardError as well here so we're certain the delete cycle runs